### PR TITLE
mgmt, azure migrate test proxy

### PR DESCRIPTION
**Pull request by weidongxu-microsoft**
There is a few tests take too long, or being too complex to get fixed in this PR.

Temporary skip the playback on them via `@DoNotRecord(skipInPlayback = true) // TODO(weidxu)`
Include LoadBalancer and VirtualNetworkGateway.

Will re-enable them in following month.

### CODE REVIEW

File: sdk/resourcemanager/azure-resourcemanager/src/samples/java/com/azure/resourcemanager/DesignPreviewSamples.java [hunk 1]
```diff
@@ -19,8 +19,8 @@ public class DesignPreviewSamples {
 
     private final String rgName = "rg-test";
     private final String contentVersion = "1.0.0.0";
-    private final String templateUri = "https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/101-cognitive-services-translate/azuredeploy.json";
-    private final String parametersUri = "https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/101-cognitive-services-translate/azuredeploy.parameters.json";
+    private final String templateUri = "https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/quickstarts/microsoft.storage/storage-account-create/azuredeploy.json";
+    private final String parametersUri = "https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/quickstarts/microsoft.storage/storage-account-create/azuredeploy.parameters.json";
```
Comment (weidongxu-microsoft) [inline sdk/resourcemanager/azure-resourcemanager/src/samples/java/com/azure/resourcemanager/DesignPreviewSamples.java:23]: use storage ARM template

File: sdk/resourcemanager/azure-resourcemanager/src/test/java/com/azure/resourcemanager/ApplicationGatewayTests.java [hunk 2]
```diff
@@ -323,29 +337,35 @@ public void testApplicationGatewaysInParallel() throws Exception {
                     .applicationGateways()
                     .define(azureResourceManager.applicationGateways().manager().resourceManager().internalContext().randomResourceName("ag", 13))
                     .withRegion(Region.US_EAST)
-                    .withNewResourceGroup(resourceGroup)
+                    .withExistingResourceGroup(resourceGroup)
                     .defineRequestRoutingRule("rule1")
-                    .fromPrivateFrontend()
+                    .fromPublicFrontend()
```
Comment (weidongxu-microsoft) [inline sdk/resourcemanager/azure-resourcemanager/src/test/java/com/azure/resourcemanager/ApplicationGatewayTests.java:342]: V2 only allow public endpoint

File: sdk/resourcemanager/azure-resourcemanager/src/test/java/com/azure/resourcemanager/KubernetesAadTests.java [hunk 3]
```diff
@@ -151,26 +151,16 @@ public void testKubernetesClusterAadIntegration() {
             Assertions.assertEquals(0, kubernetesCluster.azureActiveDirectoryGroupIds().size());
             Assertions.assertTrue(kubernetesCluster.isLocalAccountsEnabled());
 
+            // Since kubernetes version 1.25, disableLocalAccounts can only be set on Azure AD integration enabled cluster.
             kubernetesCluster.update()
+                .withAzureActiveDirectoryGroup(group.id())
                 .disableLocalAccounts()
```
Comment (weidongxu-microsoft) [inline sdk/resourcemanager/azure-resourcemanager/src/test/java/com/azure/resourcemanager/KubernetesAadTests.java:157]: AKS requires having AAD when disable local accounts.

File: sdk/resourcemanager/azure-resourcemanager/src/test/java/com/azure/resourcemanager/PrivateLinkTests.java [hunk 4]
```diff
@@ -175,31 +176,31 @@ public void testPrivateEndpoint() {
...
+//        privateEndpoint.update()
+//            .withoutPrivateLinkServiceConnection(pecName)
+//            .definePrivateLinkServiceConnection(pecName2)
+//                .withResource(storageAccount2)
+//                .withSubResource(PrivateLinkSubResourceName.STORAGE_FILE)
+//                .attach()
+//            .apply();
+//
+//        Assertions.assertEquals(Collections.singletonList(PrivateLinkSubResourceName.STORAGE_FILE), privateEndpoint.privateLinkServiceConnections().get(pecName2).subResourceNames());
+//        Assertions.assertEquals("Approved", privateEndpoint.privateLinkServiceConnections().get(pecName2).state().status());
```
Comment (weidongxu-microsoft) [inline sdk/resourcemanager/azure-resourcemanager/src/test/java/com/azure/resourcemanager/PrivateLinkTests.java:203]: this part seems no longer allowed

File: sdk/resourcemanager/azure-resourcemanager/src/test/java/com/azure/resourcemanager/TestApplicationGateway.java [hunk 5]
```diff
@@ -251,47 +253,46 @@ public void print(ApplicationGateway resource) {
...
-                                .attach()
-                                .create();
-                        }
-                    });
-
-            // Start the creation...
-            creationThread.start();
-
-            // ...But bail out after 30 sec, as it is enough to test the results
-            creationThread.join();
```
Comment (weidongxu-microsoft) [inline sdk/resourcemanager/azure-resourcemanager/src/test/java/com/azure/resourcemanager/TestApplicationGateway.java:294]: no idea why previous use thread...

File: sdk/resourcemanager/azure-resourcemanager/src/test/java/com/azure/resourcemanager/TestVirtualMachineSsh.java [hunk 6]
```diff
@@ -59,26 +54,6 @@ public VirtualMachine createResource(VirtualMachines virtualMachines) throws Exc
...
-
-        JSch jsch = new JSch();
-        Session session = null;
-        if (TestUtils.isRecordMode()) {
-            try {
-                java.util.Properties config = new java.util.Properties();
-                config.put("StrictHostKeyChecking", "no");
-                // jsch.addIdentity(sshFile, filePassword);
-                session = jsch.getSession("testuser", publicIpDnsLabel + "." + "eastus.cloudapp.azure.com", 22);
-                session.setPassword(password);
```
Comment (weidongxu-microsoft) [inline sdk/resourcemanager/azure-resourcemanager/src/test/java/com/azure/resourcemanager/TestVirtualMachineSsh.java:71]: Delete this SSH part. No idea why create VM with SSH then use password to connect.

**Review by XiaofeiCao — APPROVED** [review]

File: sdk/resourcemanager/azure-resourcemanager/src/test/java/com/azure/resourcemanager/ApplicationGatewayTests.java [hunk 7]
```diff
@@ -92,18 +96,20 @@ public void testAppGatewaysInternalComplex() throws Exception {
...
      * @throws Exception
      */
+    @Disabled("FIXME Priority is required in UrlPathMap")
     @Test
     public void testAppGatewaysPublicUrlPathBased() throws Exception {
         new TestApplicationGateway().new UrlPathBased(azureResourceManager.resourceGroups().manager().internalContext())
             .runTest(azureResourceManager.applicationGateways(), azureResourceManager.resourceGroups());
     }
 
+    @Disabled("WAF_V2 currently does not support private endpoint, and we do not want to have accessible public IP on VM. Enable it after WAF_V2 supports private endpoint.")
```
Comment (joshfree) [inline sdk/resourcemanager/azure-resourcemanager/src/test/java/com/azure/resourcemanager/ApplicationGatewayTests.java:106]: should this have a link to a tracking github issue

**Review by joshfree — APPROVED** [review]

File: sdk/resourcemanager/azure-resourcemanager/src/test/java/com/azure/resourcemanager/ApplicationGatewayTests.java (hunk 7 above)
Comment (weidongxu-microsoft) [reply sdk/resourcemanager/azure-resourcemanager/src/test/java/com/azure/resourcemanager/ApplicationGatewayTests.java:106]: Sure. Issue logged https://github.com/Azure/azure-sdk-for-java/issues/36122

**Pull request merged**